src/modules/did/did.service.ts

```diff
@@ -430,13 +442,20 @@ export class DIDService implements OnModuleInit, OnModuleDestroy {
         const cachedService = cachedServices.find(
           (claim) => claim.serviceEndpoint === serviceEndpoint
         );
-        if (cachedService) return cachedService;
+
+        if (
+          cachedService &&
+          (await this.didCluster.isPinned(serviceEndpoint))
+        ) {
+          return cachedService;
+        }
 
         if (!IPFSService.isCID(serviceEndpoint)) {
           return { serviceEndpoint, ...rest };
         }
 
         const token = await this.didStore.get(serviceEndpoint);
+        await this.didCluster.save(token);
```

Comment:
@JGiter Could you explain a little more your rationale for this approach?

I think this definitely works well, thanks for your work thus far 🙇 
However, I think we should consider if it is worth improving this a bit. My thinking is that:
- 👍  This approach ensure that every thing that `resolveNotCachedClaims` is called, that a credential which is not pinned in the new cluster will be pinned
- 👎  It unnecessarily slows down DID document resolution because there is extra pinning step, which the client doesn't really care about.
- 👎  It adds a side-effect to `resolveNotCachedClaims`. Related to this, IMO, the approach causes `resolveNotCachedClaims` violates the single responsibility principle.
-  👎 It threads the migration code through `resolveNotCachedClaims` such that it might be difficult for someone not familiar with the code to know which part to surgically remove to stop migration (they would probably need to go back and look at this PR)

What do you think of an alternative approach of creating a new private method, maybe called `pinAllServices`, that can accept an array of `IClaim` and pin to the cluster. I know this would mean potentially retrieving the credential twice from IPFS, but I think it would make the code a bit clearer. What are your thoughts?

Reply:
I agree that `resolveClaims` isn't proper place for pinning. I think pinning is more belong to caching. If so are `incrementalRefreshCachedDocument` and `addCachedDocument` more suitable for pinning?

> It unnecessarily slows down DID document resolution

Can we pin in job queue?

Reply:
> What do you think of an alternative approach of creating a new private method, maybe called pinAllServices

The queue is preferable, because it doesn't add delay to document resolution

Reply:
I think you're right, I think that `processDIDDocumentAddition` and `processDIDDocumentRefresh` will only be called from the `ADD_DID_DOC_QUEUE_NAME` and the `UPDATE_DID_DOC_QUEUE_NAME` queues. So I think we can leave it as is.